_includes/tutorials/error-handling/kstreams/markup/dev/make-exception-handler.adoc

```diff
@@ -0,0 +1,66 @@
+////
+In this file you describe the Kafka streams topology, and should cover the main points of the tutorial.
+The text assumes a method buildTopology exists and constructs the Kafka Streams application.  Feel free to modify the text below to suit your needs.
+////
+
+Before you create the Kafka Streams application you'll need to create an instance of a https://kafka.apache.org/28/javadoc/org/apache/kafka/streams/errors/StreamsUncaughtExceptionHandler[StreamsUncaughtExceptionHandler].  But before we dive into the code let's breifly cover a few points about the `StreamsUncaughtExceptionHander`.
+
+It's an important point to keep in mind that the exception handler will not work for *_all_* exceptions, just those not directly handled by Kafka Streams. An example of an exception that Kafka Streams handles is the https://kafka.apache.org/27/javadoc/org/apache/kafka/common/errors/ProducerFencedException.html[ProducerFencedException] But any exceptions related to your buisness logic are not dealt with and bubble all the way up to the `StreamThread`, leaving application no choice but shut down.  So the `StreamsUncaughtExceptionHandler` gives you a mechanism to take different actions in the case of a thrown exception, which could include shutting down, but now it's a choice you make vs. default behavior.
+
+The `StreamsUncaughtExceptionHandler` has one method `handle`, and it returns an `enum` of type https://kafka.apache.org/28/javadoc/org/apache/kafka/streams/errors/StreamsUncaughtExceptionHandler.StreamThreadExceptionResponse.html[StreamThreadExceptionResponse] which provides you the opportunity to instruct Kafka Streams how to respond to the exception.  The possible return values are:
+
+
+* REPLACE_THREAD - Replaces the thread receiving the exception and processing continues with the same number of configured threads.
```

Comment:
@bbejeck this might be a good addition here since duplicates are a big deal for some use cases.  Please note the placeholder `<insert configuration parameter name>`

```suggestion
* REPLACE_THREAD - Replaces the thread receiving the exception and processing continues with the same number of configured threads. (Note: this can result in duplicate records depending on the application's processing mode <insert configuration parameter name>)
```